### PR TITLE
1835 ShareSellingRound fixes and Bankruptcy processing

### LINKED ISSUE #326: 1856 Bridges not counted in revenue run
Hello,
Current game, 1856 with Diesel. WR has an 4-Train, Bridge, the calculation runs the train without the bridge-income. Any solution?
greetings
Volker 

[1856_20201121_1655_Kllaus-J_rgen_Fleischer.zip](https://github.com/Rails-18xx/Rails/files/5775613/1856_20201121_1655_Kllaus-J_rgen_Fleischer.zip)

### PR DESCRIPTION
**Pull request by erik-vos**
Also includes a fix for #326: 1856 Bridge/Tunnel bonus not assigned to owning company

Added new test cases.

Still to do:
- A President owning double shares only, will still get the option to sell a single share. This does not work
(and I suppose it's illegal to sell an odd number of shares in such a case).

Closes #326

### CODE REVIEW

File: src/main/java/net/sf/rails/game/financial/ShareSellingRound.java [hunk 1]
```diff
@@ -164,99 +175,136 @@ protected void findSellableShares() {
             int dumpThreshold = 0;
             SortedSet<Integer> possibleSharesToSell = null;
             boolean dumpIsPossible = false;
+            Player dumpedPlayer = company.findPlayerToDump();
+            int dumpedPlayerShares = dumpedPlayer != null
```
Comment (neutronc) [inline src/main/java/net/sf/rails/game/financial/ShareSellingRound.java:179]: Hm ? you are assigned a Player dumpedPlayer to an Int value dumpedPlayerShares and the compiler is not complaining ? Or what am i missing here ?

Comment (erik-vos) [reply src/main/java/net/sf/rails/game/financial/ShareSellingRound.java:179]: You mean this bit?
`            int dumpedPlayerShares = dumpedPlayer != null
                    ? dumpedPlayer.getPortfolioModel().getShares(company)
                    : 0;
`
That's a ?: type if/then/else. Read it as
`            int dumpedPlayerShares = (dumpedPlayer != null
                    ? dumpedPlayer.getPortfolioModel().getShares(company)
                    : 0);
`

Comment (neutronc) [reply src/main/java/net/sf/rails/game/financial/ShareSellingRound.java:179]: Missing the parantheses there

Comment (erik-vos) [reply src/main/java/net/sf/rails/game/financial/ShareSellingRound.java:179]: GitHub was misleading you by cutting off in the middle of that statement.

Comment (erik-vos) [reply src/main/java/net/sf/rails/game/financial/ShareSellingRound.java]: PR updated to add 1835 Bankruptcy handling, improbable as that may be.

Oh, and a pair of parentheses on behalf of Martin.

**Comment by erik-vos** [comment]
I have rerun the Windows installer test, that failed (or never ended) before, so I think this PR is ready to be merged.

**Pull request merged**